compiler/interface.go

```diff
@@ -169,9 +171,18 @@ func (c *compilerContext) getTypeCode(typ types.Type) llvm.Value {
 		case *types.Basic:
 			typeFields = []llvm.Value{c.getTypeCode(types.NewPointer(typ))}
 		case *types.Named:
+			name := typ.Obj().Name()
+			var buf []llvm.Value
+
+			for _, b := range []byte(name) {
+				buf = append(buf, llvm.ConstInt(c.ctx.Int8Type(), uint64(b), false))
+			}
```

Comment:
I believe the following

```go
			var buf []llvm.Value
			for _, b := range []byte(name) {
				buf = append(buf, llvm.ConstInt(c.ctx.Int8Type(), uint64(b), false))
			}
// ....
				llvm.ConstArray(c.ctx.Int8Type(), buf),
```

Is the same as the following:

```go
				c.ctx.ConstString(name, false),
```

I think the 2nd is easier to read.